### DISCUSSION TITLE
Erro na hora de realizar insert usando dapper contrib

### DISCUSSION THREAD

**Discussion by Marcos-Sousa** [andrebaltieri/flunt#109] [General]
Estou com o seguinte erro, na hora de realizar o insert usando o DAPPER.CONTRIB, pois está pegando as propriedades do Flunt.Notifications, que uso para validar as minha propriedades da classe.. como faço para não realizar esse mapeamento?
![errr](https://user-images.githubusercontent.com/25872205/187201072-a5f199d8-950d-4a76-a758-10a972c0463e.png)

**Comment by eduprog** [comment]
> Estou com o seguinte erro, na hora de realizar o insert usando o DAPPER.CONTRIB, pois está pegando as propriedades do Flunt.Notifications, que uso para validar as minha propriedades da classe.. como faço para não realizar esse mapeamento?
> ![errr](https://user-images.githubusercontent.com/25872205/187201072-a5f199d8-950d-4a76-a758-10a972c0463e.png)
> 

Se você utilizar algum tipo de mapeamento via classe como Dommel, basta usar no .Ignore no campo.

No caso de attributos, tente usar o [Computed]

http://dotnet-concept.com/Tips/2015/2/5798814/How-to-ignore-a-class-property-in-Dapper-net

https://stackoverflow.com/questions/19716568/ignore-property-on-model-property

**Reply by Marcos-Sousa** [reply]
Boa noite!! não estou utilizando nenhum tipo de mapeamento na classe, no caso estou usando somente o dapper.contrib, para realiazar o mapeamento, porém como minha classe herda do Notifiable que é um pacote de validação, na hora do insert, está tentando inserir as propriedades do Notifiable..  porém não sei como retirar essas propriedades.

**Reply by eduprog** [reply]
@Marcos-Sousa tem como colocar sua classe aqui para podermos analisar? Temos como configurar isto diretamente avisando ao dapper o que ele tem de ignorar ok.

**Reply by Marcos-Sousa** [reply]
opa!! boa tarde!!

Minha Classe é essa:
[Table("[Categoria]")]
    public class Categoria : Notifiable
    {
        public Categoria()
        {

        }
        public Categoria(string nome, string ativo)
        {
            Nome = nome;
            Ativo = ativo;

            new Contract()
             .IsNullOrEmpty(nome, "Nome", "O nome deve ser informado.")
             .HasMinLen(nome, 3, "Nome", "A nome deve conter no mínimo 3 caracteres.")
             .HasMaxLen(nome, 100, "Nome", "A nome deve conter no máximo 100 caracteres.");
        }
        public string Nome { get; private set; }
        public string Ativo { get; private set; }
    }


Meu Manipulador onde tento realizar a inserção e ocorre o erro:
![errrorrr](https://user-images.githubusercontent.com/25872205/187487102-6a641ebe-e666-4e1e-9e3e-3576a1a4a766.png)


Meu método no Repositório:

 public async Task Adicionar(Categoria categoria)
           => await _session.Connection.InsertAsync(categoria, _session.Transaction);

**Reply by eduprog** [reply]
Amigo, qual versão do Flunt vc está usando?

**Reply by Marcos-Sousa** [reply]
Desse projeto é a 1.0.4.

**Reply by eduprog** [reply]
A ok. Tô criando um projeto de teste aqui já te passo. Penso que vai precisar do extension. Só para ficar tudo ok. 

Versão do Dapper e do seu projeto? 
É net core ou Framework?

**Reply by Marcos-Sousa** [reply]
Versão do dapper(2.0.123)
Versão dapper.Contrib(2.0.78)
net core 6.

**Reply by eduprog** [reply]
@Marcos-Sousa segue um pequeno projeto que fiz, utilizando DapperExtension. Veja se vai lhe atender, pois, assim funcionou perfeitamente. 

Veja ai o teste.
[ConsoleDapperTest.zip](https://github.com/andrebaltieri/Flunt/files/9455898/ConsoleDapperTest.zip)


Instalei o DapperExtension 

```c#

public class CategoriaMapper : ClassMapper<Categoria>
{
    public CategoriaMapper()
    {

        Table("Categoria");
        Map(c => c.Ativo).Column("ativo");
        Map(c => c.Nome).Column("nome");
        Map(c => c.Notifications).Ignore();
        Map(c => c.Valid).Ignore();
        Map(c => c.Invalid).Ignore();
    }
}


 using (SqlConnection conn = new SqlConnection(connStr))
        {

            for (int i = 0; i < 20; i++)
            {
                var categoria = new Categoria($"Categoria {Guid.NewGuid().ToString("N").Substring(0,7)}", "S");
                // Verificar que este inser usado deve dapperExtension.Insert pois existe o Dapper.Contrib.Extension
                DapperExtensions.DapperExtensions.Insert<Categoria>(conn,categoria); 
            }
        }




```
Eu uso a muito tempo o Dommel que dá mais recursos e usando FluentApi como se fosse o EntityFramework, mas desta forma também funcionou normal. Ao chamar o `DapperExtensions.DapperExtensions.Insert<Categoria>(conn,categoria); `  o dapper já corre a classe e instancia e lê os mapeamentos. Caso ela esteja em uma DLL separada você terá de setar o nome do assembly. 

Algo como assim ok. 

```c#
 DapperExtensions.DapperExtensions.DefaultMapper = typeof (ServiceMapper);

 // Tell Dapper Extension to scan this assembly for custom mappings
 DapperExtensions.DapperExtensions.SetMappingAssemblies(new[]
 {
     typeof (ServiceMapper).Assembly
 });
```

Referência: https://stackoverflow.com/questions/34388946/how-to-setup-dapper-extensions-custom-mapping-with-structuremap


Espero ter ajudado.

**Reply by Marcos-Sousa** [reply]
@eduprog  muito obrigado pela resolução!!! vlw pela dica, vou dar uma analisada no código..
